### PR TITLE
docs: State that the `meltano compile` command is in beta

### PR DESCRIPTION
**Pull request by WillDaSilva**
Supersedes:
- https://github.com/meltano/meltano/pull/7321

### CODE REVIEW

**Review by aaronsteers — APPROVED** [review]
Nice 👍

**Comment by aaronsteers** [comment]
Totally optional, but @WillDaSilva , what do you think of linking to a github discussion to collect feedback and share back updates?

**Comment by WillDaSilva** [comment]
> Totally optional, but @WillDaSilva , what do you think of linking to a github discussion to collect feedback and share back updates?

@aaronsteers A GitHub discussion about what? This PR before we merge it (and whether we should require the `experimental` setting to be `true`), or something else related to the broader manifest topic?

**Comment by aaronsteers** [comment]
> > Totally optional, but @WillDaSilva , what do you think of linking to a github discussion to collect feedback and share back updates?
> 
> @aaronsteers A GitHub discussion about what? This PR before we merge it (and whether we should require the `experimental` setting to be `true`), or something else related to the broader manifest topic?

Here you go:

- https://github.com/meltano/meltano/discussions/7323

File: docs/src/_reference/command-line-interface.md [hunk 1]
```diff
@@ -122,6 +122,10 @@ The `add` command does not run relative to a [Meltano Environment](https://docs.
 
 ## `compile`
 
+<div class="notification is-warning">
+    <p>The compile command is currently experimental, and subject to change without corresponding semantic version updates.</p>
```
Comment (aaronsteers) [inline docs/src/_reference/command-line-interface.md:126]: We could add more copy or just link under the work "experimental".

```suggestion
    <p>The compile command is currently [experimental](https://github.com/meltano/meltano/discussions/7323), and subject to change without corresponding semantic version updates.</p>
```

Comment (tayloramurphy) [reply docs/src/_reference/command-line-interface.md:126]: @aaronsteers thoughts on calling it something other than _experimental_? An experiment to me indicates something we have a 50/50 shot on rolling back. Compile is something we'll very likely keep and continue to iterate on. 

Perhaps we could say the structure is not yet stable? Or is there a better phrasing here?

Comment (WillDaSilva) [reply docs/src/_reference/command-line-interface.md:126]: @tayloramurphy Thoughts on calling it a "feature preview"? Or we could say that the feature is in beta.

Comment (tayloramurphy) [reply docs/src/_reference/command-line-interface.md:126]: @WillDaSilva I guess Beta makes sense - we could also just have a 0.X.X semver on it generally with no promise on when the manifest will be 1.0, right?

Comment (WillDaSilva) [reply docs/src/_reference/command-line-interface.md:126]: I don't think we should have versions for features. Let's hope that Meltano itself never becomes so complicated as to warrant that.

Comment (tayloramurphy) [reply docs/src/_reference/command-line-interface.md:126]: @WillDaSilva sorry, not for the feature itself, but the schema of the generated file could be versioned right? https://docs.getdbt.com/reference/artifacts/manifest-json dbt-core v1.3 produces schema v7 while 1.4 produces v8. Something like that could happen, right?

Comment (WillDaSilva) [reply docs/src/_reference/command-line-interface.md:126]: @tayloramurphy The manifest and `meltano.yml` share a schema and version number. The `meltano.yml` version number is hard-coded as `1`. Ideally we won't add a manifest-specific version number, as that would mean the schema for the manifest and `meltano.yml` differ.

Comment (WillDaSilva) [reply docs/src/_reference/command-line-interface.md]: Resolved in b38571b

Comment (tayloramurphy) [reply docs/src/_reference/command-line-interface.md]: Ah ok, thanks for that context.

**Review by tayloramurphy — APPROVED** [review]

**Pull request merged**